pkg/operator/hive/hive.go

```diff
@@ -143,37 +154,45 @@ func (r *ReconcileHiveConfig) deployHive(hLog log.FieldLogger, h *resource.Helpe
 		hiveDeployment.Spec.Replicas = &replicas
 	}
 
-	result, err := h.ApplyRuntimeObject(hiveDeployment, scheme.Scheme)
-	if err != nil {
-		hLog.WithError(err).Error("error applying deployment")
-		return err
+	// Load namespaced assets, decode them, set to our target namespace, and apply:
+	coreCodecs := serializer.NewCodecFactory(scheme.Scheme)
+	coreDecoder := coreCodecs.UniversalDecoder(corev1.SchemeGroupVersion)
+	namespacedAssets := map[string]runtime.Decoder{
```

Comment:
What is the purpose of having a map with a decoder for the value when every entry in the map has the same decoder?

Reply:
When implementing, I wasn't aware it was all going to be core resources, and wanted to keep it flexible to be able to apply other things if necessary. I can't think of anything upcoming so will adjust this to not pass the decoder.

Reply:
I think that you may be incorrectly assuming that the parameter to `UniversalDecoder` does. It does not determine which apis can be decoded. Instead, it tells the decoder to use specific versions instead of the internal version. Multiple api versions can be passed to the function. If there were types in another API group, you could still use the same decoder and pass all of the api versions that you were interested in to the function. With that said, I still think that you want the deserializer instead.

Reply:
That is good to know and definitely not how I thought that worked.